File: src/main/java/com/tomatokey/framework/configuration/mvc/WebMvcConfiguration.java
package com.tomatokey.framework.configuration.mvc;

import com.tomatokey.framework.configuration.mvc.auth.AuthorizationHandlerInterceptor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebMvcConfiguration implements WebMvcConfigurer {

    @Bean
    AuthorizationHandlerInterceptor authorizationHandlerInterceptor() {
        return new AuthorizationHandlerInterceptor();
    }

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(authorizationHandlerInterceptor())
                .addPathPatterns("/**");
    }

}

File: src/main/kotlin/com/prototype/framework/configuration/mvc/WebMvcConfiguration.kt
package com.prototype.framework.configuration.mvc

import com.prototype.framework.configuration.mvc.auth.AuthorizationHandlerInterceptor
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebMvcConfiguration : WebMvcConfigurer {

    @Bean
    fun authorizationHandlerInterceptor(): AuthorizationHandlerInterceptor {
        return AuthorizationHandlerInterceptor()
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(authorizationHandlerInterceptor())
                .addPathPatterns("/**")
    }

}